### PR TITLE
feat: first draft of bazel support.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

**Comment by geoand** [comment]
I have never used Bazel so I can't really comment :P

**Comment by iocanel** [comment]
@geoand: me neither.  If you don't see something obviously wrong just approve ;-)

File: core/src/main/java/io/ap4k/project/BazelInfoReader.java [hunk 1]
```diff
@@ -35,6 +64,74 @@ public boolean isApplicable(Path root) {
 
   @Override
   public BuildInfo getInfo(Path root) {
-    throw new UnsupportedOperationException();
+    Path buildGradle = root.resolve(BUILD);
```
Comment (geoand) [inline core/src/main/java/io/ap4k/project/BazelInfoReader.java]: `buildGradle` doesn't seem like the correct name :)

File: core/src/test/java/io/ap4k/project/BazelInfoReaderTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import static org.junit.jupiter.api.Assertions.*;
+
+class BazelInfoReaderTest {
+
+  String BAZEL_INIT = "bazel/bazel-initial/BUILD";
+  String BAZEL_VERSIONED = "bazel/bazel-versioned/BUILD";
+
+  @Test
+  void shouldParsePlainBuild() {
+    URL bazelInitial = GradleInfoReaderTest.class.getClassLoader().getResource(BAZEL_INIT);
```
Comment (geoand) [inline core/src/test/java/io/ap4k/project/BazelInfoReaderTest.java]: `GradleInfoReaderTest` -> `BazelInfoReaderTest` ?

**Comment by iocanel** [comment]
@geoand: feedback had been applied.

**Comment by cmoulliard** [comment]
Can we avoid to use this horrible bazel tool ? We had this discussion with Istio's project and google finally decided to remove it

**Comment by iocanel** [comment]
@cmoulliard: we are not using bazel ourselves in any way. We are just providing a convince to people that use bazel and want to use ap4k too.

One of the strongest selling points of ap4k, is that its build tool independent. Showcasing that is really important. IMHO.

**Pull request merged**